### ISSUE TITLE
Tests fail on Windows due to EOL

### ISSUE THREAD

**Issue by jayvdb** [smhg/gettext-parser#61]
The following tests fail on Windows, very likely due to EOL only differences

```console
  PO Compiler
    Headers
      1) should keep tile casing
    UTF-8
      2) should compile
    Latin-13
      3) should compile
    Plurals
      4) should compile correct plurals in POT files
    Message folding
      5) should compile without folding
      6) should compile with different folding
    Sorting
      7) should sort output entries by msgid when `sort` is `true`
      8) should sort entries using a custom `sort` function
```

**Comment by smhg** [comment]
Thank you for the report. I've [added windows](https://github.com/smhg/gettext-parser/actions/runs/1292221992) as a test environment.
Are these the same errors as you get?
Would you be willing to submit a PR with a fix?

**Comment by jayvdb** [comment]
Yes, those are the errors.
I am willing, but have a bunch of other more important tasks atm.
I've noticed this is quite common, e.g. I raised https://github.com/BenoitZugmeyer/eslint-plugin-html/issues/147 recently.  I suspect there would be a common type of fix for doing EOL-agnostic assertions.

**Comment by smhg** [comment]
An update: I've 'fixed' the tests on Windows in a [separate branch](https://github.com/smhg/gettext-parser/tree/test-eol-win).
This means `po.compile` now uses the EOL character of the OS it runs on.

I don't know though if that's a good approach... When output is different on different OS'es, this will cause unnecessary noise (version control systems).

A solution would be to add a flag. Poedit has this. But I can't find such a thing in `xgettext`.

@jayvdb would you be able to verify what line endings `xgettext` uses on Windows? You can follow the download links on the [gettext website](https://www.gnu.org/software/gettext/).

**Comment by jayvdb** [comment]
IMO po files should be always using unix EOL and file paths.  But xgettext doesnt do this.  e.g. https://code.djangoproject.com/ticket/16493 , and there is my hacky approach to undo the Windows-isms that xgettext emits.

**Comment by smhg** [comment]
The tests use `fs.readFile` which returns data using the system EOL if I'm correct. So one option is hacky tests which convert any `\r\n` to `\n`. Correct me I'm wrong here.

But this makes me think the better approach is to make this configurable. With the system EOL as default. Making it a breaking change.

I think you are right that EOL are a PITA, but that's the world we live in.

**Comment by smhg** [comment]
I was wrong about using the system EOL by default. Like Poedit, I've set it to Unix line endings. Now there is no breaking change.
Released as `4.2.0`. Thank you for your contribution!
